src/odemis/acq/stitching/_registrar.py

```diff
@@ -580,8 +581,19 @@ def _insert_tile_to_grid(self, tile):
                     self.tiles[i].append(None)
                     self.shifts_hor[i].append(None)
                     self.shifts_ver[i].append(None)
-        else:
+        elif abs(hor_diff) == 0 and abs(ver_diff) == 0:
             raise ValueError("Cannot insert multiple tiles at the same position.")
+        elif abs(ver_diff) == abs(hor_diff):
+            # Purely diagonal shift. In practice, this will never happen because the smallest difference will be
+            # picked up by one of the previous if-statements. This exception is likely to be triggered only in
+            # artificial tests.
+            # TODO: Implement registrar for tiles which are purely diagonally shifted.
+            raise NotImplementedError("Cannot insert tiles which have a pure diagonal shift, this functionality is "
+                                      "not yet implemented. This error statement is only expected in test cases "
+                                      "since the horizontal shift needs to be exactly equal to the vertical shift.")
```

Comment:
If that only happens in some test cases, I'd suggest to fix the test cases. If for some reason you really need that, then make the error a ValueError, with a simpler message like "Horizontal and vertical shift show pure diagonal shift which is a sign of missing tile".
Also, I don't see any reason to implement code for such artificial corner case. => remove the TODO.